plugins/inputs/prometheus/kubernetes.go

```diff
@@ -126,6 +150,169 @@ func (p *Prometheus) watch(ctx context.Context, client *k8s.Client) error {
 	}
 }
 
+func (p *Prometheus) cAdvisor(ctx context.Context, client *k8s.Client) error {
+	p.Log.Infof("Using monitor pods version 2 to get pod list using cAdvisor.")
+
+	nodeIP := os.Getenv("NODE_IP")
+	if nodeIP == "" {
+		return errors.New("The environment variable NODE_IP is not set. Cannot get pod list for monitor_kubernetes_pods using version 2.")
+	}
+
+	// Parse label and field selectors - will be used to filter pods after cAdvisor call
+	labelSelector, err := labels.Parse(p.KubernetesLabelSelector)
+	if err != nil {
+		p.Log.Errorf("Error parsing the specified label selector(s): %s", err.Error())
+	}
+	fieldSelector, err := fields.ParseSelector(p.KubernetesFieldSelector)
+	if err != nil {
+		p.Log.Errorf("Error parsing the specified field selector(s): %s", err.Error())
+	}
+	if fieldSelectorIsSupported(fieldSelector) {
+		p.Log.Errorf("A specified field selector is not supported as a field selector for pods")
+	}
+	p.Log.Infof("Using the label selector: %v and field selector: %v", labelSelector, fieldSelector)
```

Comment:
let's do any config-related validation in the `func (p *Prometheus) Init() error{}` function, and return errors instead of logging them